dist/threads-shared/shared.xs

```diff
@@ -1191,7 +1209,7 @@ sharedsv_array_mg_copy(pTHX_ SV *sv, MAGIC* mg,
                             toLOWER(mg->mg_type),&sharedsv_elem_vtbl,
                             name, namlen);
     PERL_UNUSED_ARG(sv);
-    nmg->mg_flags |= MGf_DUP;
+    nmg->mg_flags |= MGf_DUP|MGf_LOCAL;
```

Comment:
Further up there's an `#ifdef MGf_LOCAL` check. Either this use of `MGf_LOCAL` needs one too, or all the `#ifdef`s should be removed.